src/main/java/org/ow2/proactive/catalogclient/service/RemoteObjectService.java

```diff
@@ -70,9 +59,14 @@ public static RemoteObjectService getInstance() {
      */
     public String getStringOnUrl(String url, String sessionId) {
         Response response = given().headers("sessionID", sessionId).when().get(url).then().extract().response();
-        ResponseUtils.checkResponse(response);
-
+        checkResponse(response);
         return response.asString();
     }
 
+    private void checkResponse(Response response) {
+        if (!(200 <= response.getStatusCode() && response.getStatusCode() <= 300)) {
+            throw new RuntimeException(response.getStatusCode() + " " + response.getStatusLine());
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)